### PR TITLE
GCE: Use WaitForOpCompletion to check if any GCE API has completed.

### PR DESCRIPTION
**Pull request by adityadani**
- All GCE APIs are asynchronous. Every API returns an Operation object.
- Query the operation object to fetch the current status of the operation.
- The operation object also returns errors and error codes if any.

### CODE REVIEW

File: gce/gce.go [hunk 1]
```diff
@@ -793,6 +823,58 @@ func (s *gceOps) waitForAttach(
...
+				if gErr, ok := err.(*googleapi.Error); ok {
+					if gErr.Code == int(404) {
+						// operation does not exist
+						return nil, false, nil
+					}
+				}
+				return nil, true, fmt.Errorf("failed to query gce operation %v for %v: %v", operation.Name, cloudopsOperationName, err)
+			}
+
+			if op == nil || op.Status != doneStatus {
```
Comment (piyush-nimbalkar) [inline gce/gce.go]: Even for failed operations, the status is DONE?

File: gce/gce.go [hunk 2]
```diff
@@ -598,8 +625,11 @@ func (s *gceOps) Snapshot(
 }
 
 func (s *gceOps) SnapshotDelete(snapID string) error {
-	_, err := s.computeService.Snapshots.Delete(s.inst.project, snapID).Do()
-	return err
+	operation, err := s.computeService.Snapshots.Delete(s.inst.project, snapID).Do()
+	if err != nil {
+		return err
+	}
+	return s.waitForOpCompletion("SnapshotDelete", s.inst.zone, operation)
```
Comment (piyush-nimbalkar) [inline gce/gce.go]: *snapshot.Delete?

File: gce/gce.go [hunk 3]
```diff
@@ -580,11 +603,15 @@ func (s *gceOps) Snapshot(
 		Name: fmt.Sprintf("snap-%d%02d%02d", time.Now().Year(), time.Now().Month(), time.Now().Day()),
 	}
 
-	_, err := s.computeService.Disks.CreateSnapshot(s.inst.project, s.inst.zone, disk, rb).Do()
+	operation, err := s.computeService.Disks.CreateSnapshot(s.inst.project, s.inst.zone, disk, rb).Do()
 	if err != nil {
 		return nil, err
 	}
 
+	if opErr := s.waitForOpCompletion("disk.Snapshot", s.inst.zone, operation); opErr != nil {
```
Comment (piyush-nimbalkar) [inline gce/gce.go]: *disk.CreateSnapshot?

File: gce/gce.go (hunk 1 above)
Comment (adityadani) [reply gce/gce.go]: Yes that operation status is set to DONE and the error is updated if any.

**Pull request merged**